### PR TITLE
VAULT: Add check that depositor cannot be vault account

### PR DESCRIPTION
**Pull request by coachchucksol**
Ensures that the vault cannot be the depositor, which would dilute the VRT supply

### CODE REVIEW

**Comment by coachchucksol** [comment]
@buffalu or @ebatsell know anyway to skip lines in Codecov? There is no way to test the added checks since they only future-proof the program

File: vault_program/src/mint_to.rs [hunk 1]
```diff
@@ -69,6 +71,16 @@ pub fn process_mint(
...
+    }
+
+    if depositor_token_account.key.eq(vault_token_account.key) {
+        msg!("Depositor token account cannot be the vault token account");
+        return Err(VaultError::InvalidDepositTokenAccount.into());
+    }
+
     let MintSummary {
         vrt_to_depositor,
         vrt_to_fee_wallet,
```
Comment (buffalu) [inline vault_program/src/mint_to.rs:86]: can you also add checks after the invokes to ensure the vault mint amount = the token mint supply + other things mentioned in the audit report?

File: vault_program/src/mint_to.rs [hunk 2]
```diff
@@ -69,6 +71,16 @@ pub fn process_mint(
     vault.check_vrt_mint(vrt_mint.key)?;
     vault.check_update_state_ok(Clock::get()?.slot, config.epoch_length())?;
 
+    if depositor.key.eq(vault_info.key) {
```
[the remark this answers is not in the document]
Comment (coachchucksol) [reply vault_program/src/mint_to.rs:74]: 10-4

File: vault_program/src/mint_to.rs (hunk 1 above)
Comment (coachchucksol) [reply vault_program/src/mint_to.rs:86]: 10-4

File: vault_program/src/mint_to.rs [hunk 3]
```diff
@@ -93,6 +114,21 @@ pub fn process_mint(
         )?;
     }
 
+    let vault_amount_after = {
```
Comment (buffalu) [inline vault_program/src/mint_to.rs:117]: imo worth moving the checks to below the transfer + mint invokes. then do:

assert(vault.vrt_supply == vrt_mint.total_supply);
assert(vault.tokens_deposited == token_account.amount);

Comment (buffalu) [reply vault_program/src/mint_to.rs]: ah yeah that makes sense, good call

**Review by buffalu — APPROVED** [review]

**Review by ebatsell — APPROVED** [review]

**Review by buffalu — APPROVED** [review]

**Pull request merged**